packages/compass/src/main/window-manager.ts

```diff
@@ -0,0 +1,313 @@
+/**
+ * A high-level wrapper around electron's builtin [BrowserWindow][0] class.
+ * https://github.com/atom/electron/blob/main/docs/api/browser-window.md
+ */
+import { pathToFileURL } from 'url';
+import path from 'path';
+import createDebug from 'debug';
+import { ipcMain } from 'hadron-ipc';
+import { once } from 'events';
+import {
+  app as electronApp,
+  shell,
+  dialog,
+  BrowserWindow,
+  BrowserWindowConstructorOptions,
+  FindInPageOptions,
+} from 'electron';
+import COMPASS_ICON from './icon';
+import { CompassMenu } from './menu';
+import type { CompassApplication } from './application';
+
+const debug = createDebug('mongodb-compass:electron:window-manager');
+
+/**
+ * Constants for window sizes on multiple platforms
+ */
+
+/**
+ * The outer dimensions to use for new windows.
+ */
+const DEFAULT_WIDTH = 1280;
+const DEFAULT_HEIGHT = (() => {
+  let height = 840;
+  /**
+   * Adjust the heights to account for platforms
+   * that use a single menu bar at the top of the screen.
+   */
+  if (process.platform === 'linux') {
+    height -= 30;
+  } else if (process.platform === 'darwin') {
+    height -= 60;
+  }
+  return height;
+})();
+
+const MIN_WIDTH = 1024;
+const MIN_HEIGHT = 640;
+
+/**
+ * The app's HTML shell which is the output of `./src/index.html`
+ * created by the `build:pages` gulp task.
+ */
+const DEFAULT_URL =
+  process.env.COMPASS_INDEX_RENDERER_URL ||
+  pathToFileURL(path.join(__dirname, 'index.html')).toString();
+
+// track if app was launched, @see `renderer ready` handler below
+let appLaunched = false;
+
+async function showWindowWhenReady(bw: BrowserWindow) {
+  await once(bw, 'ready-to-show');
```

Comment:
Is it possible that ready-to-show could have already been emitted before this function is called?

Reply:
Nevermind I see this is indeed called before window.loadURL which I assume is where that will happen.